pipelines/split/bin.yaml

```diff
@@ -0,0 +1,40 @@
+name: Split executable files
+
+needs:
+  packages:
+    - busybox
+
+inputs:
+  package:
+    description: |
+      The package to split executable files from
+    required: false
+
+pipeline:
+  - runs: |
+      pkgname="${{context.name}}"
+      pkgnobin="${pkgname%%-bin}"
+      pkgnobindir="${{targets.outdir}}/$pkgnobin"
+
+      if [ -n "${{inputs.package}}" ]; then
+        PACKAGE_DIR="${{targets.outdir}}/${{inputs.package}}"
+      elif [ "$pkgname" != "$pkgnobin" ] && [ -d "$pkgnobindir" ]; then
+        PACKAGE_DIR="$pkgnobindir"
+      else
+        PACKAGE_DIR="${{targets.outdir}}/${{targets.destdir}}"
+      fi
+
+      if [ "$PACKAGE_DIR" == "${{targets.contextdir}}" ]; then
+        echo "ERROR: Package can not split files from itself!" && exit 1
```

Comment:
don't && exit 1 here. just exit 1 on a new line. if the echo fails for some reason, then you will actually go _not_ exit here.
shell programming is madness . https://smoser.github.io/2023/04/27/set-e-bad.html

```
$ sh -ec 'false && echo exit 1; echo still here'
still here
```